filament/src/PostProcessManager.cpp

```diff
@@ -1353,45 +1380,53 @@ FrameGraphId<FrameGraphTexture> PostProcessManager::bloomPass(FrameGraph& fg,
                 mi->setParameter("invHighlight", std::isinf(bloomOptions.highlight) ? 0.0f : 1.0f / bloomOptions.highlight);
 
                 for (size_t i = 0; i < bloomOptions.levels; i++) {
-                    auto hwOutRT = resources.get(data.outRT[i]);
+                    const bool parity = (i % 2) == 0;
```

Comment:
does this code work if the parity if `bloomOptions.levels` changes?

Reply:
Yes, I had to think about this: the parity of the downsample pass is straightforward but the parity of the upsample is adjusted accordingly.